refactored API

Debrej opened the pull request.

Debrej committed:
feat: refactoring populating script
- changes to the promises flow to help reading it
- adding individual files for each model to ease adding and removing elements

Debrej committed:
setup: adding env variables to setup script

Debrej committed:
refactor: complete refactor of the requests
- using express router instead of app
- readability generally enhanced
- better practices with express or sequelize

Debrej committed:
refactor: updating authorization params

Debrej committed:
docker: adding a variable port for api db

to remove in production

Debrej committed:
refactor: changing file to import the routers

Azmah-Bad committed:
[DEPOPULATE] 👪 fixed depopulate.js script
The script now frop the database and create another one making it faster and more reliable

Azmah-Bad committed:
[SCRIPT] 📖 added repopulate
in development and testing the cammand `npm run depopulate` can be useful to create a fresh instance of the database

Azmah-Bad committed:
[SCRIPT] 🎬 added some protection to the production
script can't be run on production

Debrej commented on setup/create_admin.sh:
```diff
@@ -2,6 +2,8 @@
 # DO NOT RUN THIS SCRIPT ON PRODUCTION
 # Create a user with admin role in keycloak to make it easier to test, CI and quickstart the project
 
+export $(xargs < .env)
```
@Azmah-Bad si on est dans un env qui n'as pas encore ses variables d'environnements set il faut qu'on les importes

Azmah-Bad replied:
yes ca été solved

Debrej commented:
@Azmah-Bad check la façon dont on fait les requests et comment on populate la database, j'ai rework le flow de promesses

Azmah-Bad committed:
Merge branch 'master' into thibaut-dev

# Conflicts:
#	.example.env
#	docker-compose.yml

Azmah-Bad committed:
[PROD] 🔨 fixed database port in production

Azmah-Bad commented:
@Debrej pourquoi dans les routes de l'api tu exporte une fonction qui retourne un `Router`, ca serai pas mieux d'export que le `Router` ca revient a la meme chose mais plus clean

Azmah-Bad commented:
ok c'est pas mal les modifications que t'as fait au routes c'est mieux comme ca je vais changer les tests de la CI pour que ca marche avec tes nouveaux routes.

Azmah-Bad committed:
(refactor): 📖 made routes export a router
instead of exporting a function that returns the router the files now export the router directly, this was possible by splitting the index.js into 2 files app.js that export all the necessary modules to run the api and the index.js that only runs the runs the api. 🥳

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
(refactor): 📖 corrected http codes
POST gives back 201 `created`
PUT gives back 202 `accepted`

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
(refactor): 🚀 adapted CI to new routes

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
(refactor): 🔨 fixed event route to return code 201
mistakes were made... the event route was returning a code 2021 instead of 201

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
(refactor): 🔨 fixed CI routes

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
(refactor): 🔨 fixed CI routes tasks and shifts

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad committed:
Merge branch 'hamza-dev' into thibaut-dev

# Conflicts:
#	setup/populate.js

Azmah-Bad commented:
populate script not working, got the following error: 
```javascript
{
    code: 'ER_NO_REFERENCED_ROW_2',
    errno: 1452,
    sqlState: '23000',
    sqlMessage: 'Cannot add or update a child row: a foreign key constraint fails (`project_a`.`activities`, CONSTRAINT `activities_ibfk_1` FOREIGN KEY (`supervisor_id`) REFERENCES `users` (`id`) ON DELETE SET NULL ON UPDATE CASCADE)',
    sql: 'INSERT INTO `activities` (`id`,`name`,`description`,`start_date`,`end_date`,`created_at`,`updated_at`,`supervisor_id`,`event_id`) VALUES (NULL,\'Ramener le fromage à la maison\',\'on a acheté bcp bcp (bcp) de fromage et on doit le stocker dans le frigos\',\'2019-05-11 12:00:00\',\'2019-05-11 18:00:00\',\'2020-12-20 21:25:21\',\'2020-12-20 21:25:21\',2,1),(NULL,\'Ranger le QG orga\',\'la manif est fini c\\"était bien rigolo mais maintenant au boulot on range tout\',\'2019-05-19 20:00:00\',\'2019-05-20 18:00:00\',\'2020-12-20 21:25:21\',\'2020-12-20 21:25:21\',3,1);',
    parameters: undefined
  },
  sql: 'INSERT INTO `activities` (`id`,`name`,`description`,`start_date`,`end_date`,`created_at`,`updated_at`,`supervisor_id`,`event_id`) VALUES (NULL,\'Ramener le fromage à la maison\',\'on a acheté bcp bcp (bcp) de fromage et on doit le stocker dans le frigos\',\'2019-05-11 12:00:00\',\'2019-05-11 18:00:00\',\'2020-12-20 21:25:21\',\'2020-12-20 21:25:21\',2,1),(NULL,\'Ranger le QG orga\',\'la manif est fini c\\"était bien rigolo mais maintenant au boulot on range tout\',\'2019-05-19 20:00:00\',\'2019-05-20 18:00:00\',\'2020-12-20 21:25:21\',\'2020-12-20 21:25:21\',3,1);',
  parameters: undefined,
  fields: [ 'supervisor_id' ],
  table: 'users',
  value: undefined,
  index: 'activities_ibfk_1',
  reltype: 'child'
}
```

Debrej commented:
@Azmah-Bad parcequ'il faut passer des arguments au fichier

Azmah-Bad committed:
(refactor): 🔨 fixed populate script
and added async await syntax for cleaner and more readable code

Signed-off-by: Azmah-Bad <56384031+Azmah-Bad@users.noreply.github.com>

Azmah-Bad reviewed (approved):
![image](https://user-images.githubusercontent.com/56384031/102793039-97306f80-43a9-11eb-9352-c49ff434265d.gif)
